jobs/async-upload/job/download.py

```diff
@@ -74,6 +95,79 @@ def download_from_hf(uri: str, dest_dir: str) -> str:
     )
 
 
+def download_from_http(uri: str, dest_dir: str) -> str:
+    """
+    adapted from kserve:
+    https://github.com/kserve/kserve/blob/4edbb36c520c2e880842229bfc56b7f11d766822/python/storage/kserve_storage/kserve_storage.py#L698-L771
+    """
+    url = urlparse(uri)
+    filename = os.path.basename(url.path)
+    # Determine if the symbol '?' exists in the path
+    if mimetypes.guess_type(url.path)[0] is None and url.query != "":
+        mimetype, encoding = mimetypes.guess_type(url.query)
+    else:
+        mimetype, encoding = mimetypes.guess_type(url.path)
+
+    if filename == "":
+        raise ValueError(f"No filename contained in URI: {uri}")
+
+    # Get header information from host url
+    host_uri = url.hostname
+    headers_env_name = host_uri + HEADERS_SUFFIX
+    if headers_env_name in os.environ:
+        headers = json.loads(os.environ[headers_env_name])
+    else:
+        headers = {}
+
+    # Use body content workflow to defer body download until response.raw is called
+    # https://requests.readthedocs.io/en/latest/user/advanced/#body-content-workflow
+    with requests.get(uri, stream=True, headers=headers) as response:
+        response.raise_for_status()
+
+        is_archive = True
+        match mimetype:
+            case "application/zip":
+                valid_content_types = ZIP_CONTENT_TYPES
+            case "application/x-tar":
+                valid_content_types = TAR_CONTENT_TYPES
+            case _:
+                valid_content_types = REGULAR_FILE_CONTENT_TYPES
+                is_archive = False
```

Comment:
This is a simplified version of what we have in the [kserve-storage function](https://github.com/kserve/kserve/blob/4edbb36c520c2e880842229bfc56b7f11d766822/python/storage/kserve_storage/kserve_storage.py#L698-L771). I wonder if we need to be this strict or if we should simply download whatever url we get. Maybe a warning instead of an error would make sense.